diff -du proto/mbo/testing/proto/proto_test_util.cc proto-matchers/mbo/testing/proto/proto_test_util.cc
--- proto/mbo/testing/proto/proto_test_util.cc	2023-07-15 17:57:58
+++ proto/mbo/testing/proto/proto_test_util.cc	2023-07-15 17:47:11
@@ -14,24 +14,24 @@
 
 // READ: https://google.github.io/cpp-proto-builder
 
-#include "proto_builder/oss/testing/proto_test_util.h"
+#include "mbo/testing/proto/proto_test_util.h"
 
 #include <algorithm>
 #include <string>
+#include <string_view>
 
-#include "google/protobuf/io/tokenizer.h"
+#include "absl/log/absl_check.h"
+#include "absl/log/absl_log.h"
+#include "absl/strings/substitute.h"
 #include "gmock/gmock.h"
+#include "google/protobuf/io/tokenizer.h"
 #include "gtest/gtest.h"
-#include "absl/strings/substitute.h"
 #include "re2/re2.h"
 
-namespace testing {
-namespace oss {
-
+namespace mbo::testing::proto {
 namespace internal {
 
-using absl::string_view;
-using RegExpStringPiece = ::re2::StringPiece;
+using RegExpStringPiece = std::string_view;
 
 // Utilities.
 
@@ -71,10 +71,10 @@
 }
 
 template <typename Container>
-std::string JoinStringPieces(const Container& strings, string_view separator) {
+std::string JoinStringPieces(const Container& strings, std::string_view separator) {
   std::stringstream stream;
-  absl::string_view sep = "";
-  for (const absl::string_view str : strings) {
+  std::string_view sep = "";
+  for (const std::string_view str : strings) {
     stream << sep << str;
     sep = separator;
   }
@@ -86,7 +86,7 @@
     const ::google::protobuf::Descriptor* proto_descriptor,
     const std::vector<std::string>& ignore_fields) {
   std::vector<const ::google::protobuf::FieldDescriptor*> ignore_descriptors;
-  std::vector<absl::string_view> remaining_descriptors;
+  std::vector<std::string_view> remaining_descriptors;
 
   const ::google::protobuf::DescriptorPool* pool = proto_descriptor->file()->pool();
   for (const std::string& name : ignore_fields) {
@@ -97,7 +97,7 @@
     }
   }
 
-  QCHECK(remaining_descriptors.empty())
+  ABSL_QCHECK(remaining_descriptors.empty())
       << "Could not find fields for proto " << proto_descriptor->full_name()
       << " with fully qualified names: "
       << JoinStringPieces(remaining_descriptors, ",");
@@ -197,7 +197,7 @@
   while (!input.empty()) {
     // Consume a dot, except on the first iteration.
     if (input.size() < relative_field_path.size() && !Consume(&input, ".")) {
-      LOG(FATAL) << "Cannot parse field path '" << relative_field_path
+      ABSL_LOG(FATAL) << "Cannot parse field path '" << relative_field_path
                  << "' at offset " << relative_field_path.size() - input.size()
                  << ": expected '.'";
     }
@@ -208,41 +208,41 @@
         RE2::Consume(&input, field_regex, &name)) {
       if (field_path.empty()) {
         field.field = root_descriptor.FindFieldByName(std::string(name));
-        CHECK(field.field) << "No such field '" << name << "' in message '"
+        ABSL_CHECK(field.field) << "No such field '" << name << "' in message '"
                            << root_descriptor.full_name() << "'";
       } else {
         const ::google::protobuf::util::MessageDifferencer::SpecificField& parent =
             field_path.back();
         field.field = parent.field->message_type()->FindFieldByName(std::string(name));
-        CHECK(field.field) << "No such field '" << name << "' in '"
+        ABSL_CHECK(field.field) << "No such field '" << name << "' in '"
                            << parent.field->full_name() << "'";
       }
     } else if (RE2::Consume(&input, extension_regex, &name)) {
       field.field =
           ::google::protobuf::DescriptorPool::generated_pool()->FindExtensionByName(name);
-      CHECK(field.field) << "No such extension '" << name << "'";
+      ABSL_CHECK(field.field) << "No such extension '" << name << "'";
       if (field_path.empty()) {
-        CHECK(root_descriptor.IsExtensionNumber(field.field->number()))
+        ABSL_CHECK(root_descriptor.IsExtensionNumber(field.field->number()))
             << "Extension '" << name << "' does not extend message '"
             << root_descriptor.full_name() << "'";
       } else {
         const ::google::protobuf::util::MessageDifferencer::SpecificField& parent =
             field_path.back();
-        CHECK(parent.field->message_type()->IsExtensionNumber(
+        ABSL_CHECK(parent.field->message_type()->IsExtensionNumber(
             field.field->number()))
             << "Extension '" << name << "' does not extend '"
             << parent.field->full_name() << "'";
       }
     } else {
-      LOG(FATAL) << "Cannot parse field path '" << relative_field_path
+      ABSL_LOG(FATAL) << "Cannot parse field path '" << relative_field_path
                  << "' at offset " << relative_field_path.size() - input.size()
                  << ": expected field or extension";
     }
     field_path.push_back(field);
   }
 
-  CHECK(!field_path.empty());
-  CHECK(field_path.back().index == -1)
+  ABSL_CHECK(!field_path.empty());
+  ABSL_CHECK(field_path.back().index == -1)
       << "Terminally ignoring fields by index is currently not supported ('"
       << relative_field_path << "')";
   return field_path;
@@ -382,5 +382,4 @@
 }
 
 }  // namespace internal
-}  // namespace oss
-}  // namespace testing
+}  // namespace mbo::testing::proto
diff -du proto/mbo/testing/proto/proto_test_util.h proto-matchers/mbo/testing/proto/proto_test_util.h
--- proto/mbo/testing/proto/proto_test_util.h	2023-07-15 17:57:58
+++ proto/mbo/testing/proto/proto_test_util.h	2023-07-15 17:41:47
@@ -14,21 +14,17 @@
 
 // READ: https://google.github.io/cpp-proto-builder
 
-#ifndef PROTO_BUILDER_OSS_TESTING_PROTO_TEST_UTIL_H_
-#define PROTO_BUILDER_OSS_TESTING_PROTO_TEST_UTIL_H_
+#ifndef MBO_TESTING_PROTO_PROTO_TEST_UTIL_H_
+#define MBO_TESTING_PROTO_PROTO_TEST_UTIL_H_
 
+#include "absl/log/absl_check.h"
+#include "gmock/gmock.h"
 #include "google/protobuf/message.h"
 #include "google/protobuf/text_format.h"
 #include "google/protobuf/util/message_differencer.h"
-#include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
-// Last include to override Macros
-#include "proto_builder/oss/logging.h"
-
-namespace testing {
-namespace oss {
-
+namespace mbo::testing::proto {
 namespace internal {
 
 // Utilities.
@@ -205,7 +201,7 @@
 
   // Sets the margin of error for approximate floating point comparison.
   void SetMargin(double margin) {
-    CHECK_GE(margin, 0.0) << "Using a negative margin for Approximately";
+    ABSL_CHECK_GE(margin, 0.0) << "Using a negative margin for Approximately";
     comp_->has_custom_margin = true;
     comp_->float_margin = margin;
   }
@@ -213,7 +209,7 @@
   // Sets the relative fraction of error for approximate floating point
   // comparison.
   void SetFraction(double fraction) {
-    CHECK(0.0 <= fraction && fraction < 1.0)
+    ABSL_CHECK(0.0 <= fraction && fraction < 1.0)
         << "Fraction for Approximately must be >= 0.0 and < 1.0";
     comp_->has_custom_fraction = true;
     comp_->float_fraction = fraction;
@@ -317,7 +313,7 @@
       : ProtoMatcherBase(must_be_initialized, comp),
         expected_(CloneProto2(expected)) {
     if (must_be_initialized) {
-      CHECK(expected.IsInitialized())
+      ABSL_CHECK(expected.IsInitialized())
           << "The protocol buffer given to *InitializedProto() "
           << "must itself be initialized, but the following required fields "
           << "are missing: " << expected.InitializationErrorString() << ".";
@@ -586,7 +582,7 @@
 
   // Sets the margin of error for approximate floating point comparison.
   void SetMargin(double margin) {
-    CHECK_GE(margin, 0.0) << "Using a negative margin for Approximately";
+    ABSL_CHECK_GE(margin, 0.0) << "Using a negative margin for Approximately";
     comp_->has_custom_margin = true;
     comp_->float_margin = margin;
   }
@@ -594,7 +590,7 @@
   // Sets the relative fraction of error for approximate floating point
   // comparison.
   void SetFraction(double fraction) {
-    CHECK(0.0 <= fraction && fraction <= 1.0)
+    ABSL_CHECK(0.0 <= fraction && fraction <= 1.0)
         << "Fraction for Relatively must be >= 0.0 and < 1.0";
     comp_->has_custom_fraction = true;
     comp_->float_fraction = fraction;
@@ -820,7 +816,6 @@
           ::testing::SafeMatcherCast<const Proto&>(inner_matcher)));
 }
 
-}  // namespace oss
-}  // namespace testing
+}  // namespace mbo::testing::proto
 
-#endif  // PROTO_BUILDER_OSS_TESTING_PROTO_TEST_UTIL_H_
+#endif  // MBO_TESTING_PROTO_PROTO_TEST_UTIL_H_
