diff -du ../cpp-proto-builder/proto_builder/oss/parse_text_proto.cc proto/mbo/proto/parse_text_proto.cc
--- ../cpp-proto-builder/proto_builder/oss/parse_text_proto.cc	2023-07-15 17:24:10
+++ proto/mbo/proto/parse_text_proto.cc	2023-07-15 20:58:22
@@ -1,4 +1,5 @@
 // Copyright 2021 The CPP Proto Builder Authors
+// Copyright 2023 M.Boerger
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -12,10 +13,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// READ: https://google.github.io/cpp-proto-builder
+#include "mbo/proto/parse_text_proto.h"
 
-#include "proto_builder/oss/parse_text_proto.h"
-
+#include <source_location>
 #include <string>
 #include <vector>
 
@@ -25,7 +25,7 @@
 #include "absl/strings/str_format.h"
 #include "absl/strings/string_view.h"
 
-namespace proto_builder::oss {
+namespace mbo::proto {
 namespace internal {
 namespace {
 
@@ -80,20 +80,18 @@
 
 }  // namespace
 
-absl::Status ParseTextInternal(absl::string_view text_proto, Message* message,
-                               SourceLocation loc) {
+absl::Status ParseTextInternal(std::string_view text_proto, ::google::protobuf::Message* message,
+                               std::source_location loc) {
   google::protobuf::TextFormat::Parser parser;
   SilentErrorCollector error_collector;
   parser.RecordErrorsTo(&error_collector);
   if (parser.ParseFromString(std::string(text_proto), message)) {
     return absl::OkStatus();
   }
-  // TODO When SourceLocation becomes open source, pass loc to error
-  // status. Until then, append loc to error message.
   return absl::InvalidArgumentError(
       absl::StrFormat("%s\nFile: '%s', Line: %d", error_collector.GetErrors(),
                       loc.file_name(), loc.line()));
 }
 
 }  // namespace internal
-}  // namespace proto_builder::oss
+}  // namespace mbo::proto
diff -du ../cpp-proto-builder/proto_builder/oss/parse_text_proto.h proto/mbo/proto/parse_text_proto.h
--- ../cpp-proto-builder/proto_builder/oss/parse_text_proto.h	2023-07-15 17:24:10
+++ proto/mbo/proto/parse_text_proto.h	2023-07-15 20:56:57
@@ -1,4 +1,5 @@
 // Copyright 2021 The CPP Proto Builder Authors
+// Copyright 2023 M.Boerger
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -12,54 +13,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// READ: https://google.github.io/cpp-proto-builder
+#ifndef MBO_PROTO_PARSE_TEXT_PROTO_H_
+#define MBO_PROTO_PARSE_TEXT_PROTO_H_
 
-#ifndef PROTO_BUILDER_OSS_PARSE_TEXT_PROTO_H_
-#define PROTO_BUILDER_OSS_PARSE_TEXT_PROTO_H_
-
+#include <source_location>
 #include <string>
+#include <string_view>
 
-#include "proto_builder/oss/qcheck.h"
-#include "proto_builder/oss/source_location.h"
-#include "google/protobuf/message.h"
+#include "absl/log/absl_check.h"
 #include "absl/status/status.h"
 #include "absl/strings/str_format.h"
 #include "absl/strings/string_view.h"
+#include "google/protobuf/message.h"
 
-namespace proto_builder::oss {
-
-using ::google::protobuf::Message;
-
+namespace mbo::proto {
 namespace internal {
 
-absl::Status ParseTextInternal(absl::string_view text_proto, Message* message,
-                               SourceLocation loc);
+absl::Status ParseTextInternal(std::string_view text_proto, ::google::protobuf::Message* message,
+                               std::source_location loc);
 
 }  // namespace internal
 
 // Parses the text in 'text_proto' into a prot message of type 'T'.
 // The function dies if parsing fails.
 template <typename T>
-T ParseTextOrDie(absl::string_view text_proto,
-                 SourceLocation loc = SourceLocation::current()) {
+T ParseTextOrDie(std::string_view text_proto,
+                 std::source_location loc = std::source_location::current()) {
   T message;
   absl::Status result = internal::ParseTextInternal(text_proto, &message, loc);
-  QCheck(
-      result.ok(),
-      absl::StrFormat("ParseTextOrDie<%s>", T::GetDescriptor()->name()).c_str(),
-      QCheckOptions::kDefault, loc)
-      << result;
+  ABSL_CHECK_OK(result) << "ParseTextOrDie<" << T::GetDescriptor()->name() << ">"
+      << " @" << loc.file_name() << ":" << loc.line() << ": " << result;
   return message;
 }
 
 class ParseTextProtoHelper final {
  public:
-  ParseTextProtoHelper(absl::string_view text_proto, SourceLocation loc)
+  ParseTextProtoHelper(std::string_view text_proto, std::source_location loc)
       : text_proto_(text_proto), loc_(loc), parsed_(false) {}
 
   ~ParseTextProtoHelper() {
-    QCheck(parsed_,
-           absl::StrFormat("ParseTextProtoOrDie<T> result unused").c_str());
+    ABSL_CHECK(parsed_) << "ParseTextProtoOrDie<T> result unused";
   }
 
   template <typename T>
@@ -70,16 +63,16 @@
 
  private:
   const std::string text_proto_;
-  const SourceLocation loc_;
+  const std::source_location loc_;
   bool parsed_;
 };
 
 inline ParseTextProtoHelper ParseTextProtoOrDie(
-    absl::string_view text_proto,
-    SourceLocation loc = SourceLocation::current()) {
+    std::string_view text_proto,
+    std::source_location loc = std::source_location::current()) {
   return ParseTextProtoHelper(text_proto, loc);
 }
 
-}  // namespace proto_builder::oss
+}  // namespace mbo::proto
 
-#endif  // PROTO_BUILDER_OSS_PARSE_TEXT_PROTO_H_
+#endif  // MBO_PROTO_PARSE_TEXT_PROTO_H_
diff -du ../cpp-proto-builder/proto_builder/oss/parse_text_proto_test.cc proto/mbo/proto/parse_text_proto_test.cc
--- ../cpp-proto-builder/proto_builder/oss/parse_text_proto_test.cc	2023-07-15 17:24:10
+++ proto/mbo/proto/parse_text_proto_test.cc	2023-07-15 21:04:22
@@ -1,4 +1,5 @@
 // Copyright 2021 The CPP Proto Builder Authors
+// Copyright 2023 M.Boerger
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -12,18 +13,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// READ: https://google.github.io/cpp-proto-builder
+#include "mbo/proto/parse_text_proto.h"
 
-#include "proto_builder/oss/parse_text_proto.h"
-
-#include "proto_builder/oss/tests/simple_message.pb.h"
 #include "gmock/gmock.h"
-#include "proto_builder/oss/testing/cpp_pb_gunit.h"
+#include "mbo/proto/tests/simple_message.pb.h"
+#include "mbo/testing/proto/proto_matchers.h"
 
-namespace proto_builder::oss {
+namespace mbo::proto {
 namespace {
 
-using ::testing::oss::EqualsProto;
+using ::mbo::testing::proto::EqualsProto;
+using ::mbo::proto::tests::SimpleMessage;
 
 class ParseTextProtoTest : public ::testing::Test {};
 
@@ -35,12 +35,13 @@
 }
 
 TEST_F(ParseTextProtoTest, ParseError) {
-  EXPECT_DEATH(ParseTextOrDie<oss::SimpleMessage>("!!!"),
+  EXPECT_DEATH(ParseTextOrDie<SimpleMessage>("!!!"),
                // Using [0-9] in lieu of \\d to be compatible in open source.
-               "File: '.*/parse_text_proto_test.cc', Line: [0-9]+\n"
-               "Check failed: ParseTextOrDie<SimpleMessage>\n"
+               ".*Check failed:.*\n*"
+               "File: '.*/parse_text_proto.*', Line: [0-9]+.*"
+               "ParseTextOrDie<SimpleMessage>.*"
                "INVALID_ARGUMENT: Line 0, Col 0: Expected identifier, got: !");
 }
 
 }  // namespace
-}  // namespace proto_builder::oss
+}  // namespace mbo::proto
